python/afdko/buildcff2vf.py

```diff
@@ -2,996 +2,498 @@
+        if options.verbose:
+            level = logging.INFO
+        logging.basicConfig(level=level)
+    logger.setLevel(level)
+    return options
 
 
 def run(args=None):
-    post_format_3 = False
-    if not args:
+    def otfFinder(s):
+        return s.replace('.ufo', options.source_font_suffix)
+
+    if args is None:
         args = sys.argv[1:]
-    if '-u' in args:
-        print(__usage__)
-        return
-    if '-h' in args:
-        print(__help__)
-        return
-    if '-p' in args:
-        post_format_3 = True
-        args.remove('-p')
 
-    if parse_version(fontToolsVersion) < parse_version("3.19"):
-        print("Quitting. The Python fonttools module must be at least 3.19.0 "
-              "in order for buildcff2vf to work.")
-        return
+    options = get_options(args)
 
-    if len(args) == 2:
-        designSpacePath, varFontPath = args
-    elif len(args) == 1:
-        designSpacePath = args[0]
-        varFontPath = os.path.splitext(designSpacePath)[0] + '.otf'
-    else:
-        print(__usage__)
+    if parse_version(fontToolsVersion) < parse_version("3.37"):
+        logger.error("Quitting. The Python fonttools module "
+                     "must be at least 3.37.")
         return
 
-    if os.path.exists(varFontPath):
-        os.remove(varFontPath)
-    varFont, varModel, masterPaths = varLib.build(designSpacePath,
-                                                  otfFinder, exclude=("CFF2",))
-
-    blendError = buildCFF2Font(varFontPath, varFont, varModel, masterPaths,
-                               post_format_3)
-    if not blendError:
-        print("Built variable font '%s'" % (varFontPath))
+    if os.path.exists(options.var_font_path):
+        os.remove(options.var_font_path)
+
+    designspace = DesignSpaceDocument.fromfile(options.design_space_path)
+    ds_data = varLib.load_designspace(designspace)
+    master_fonts = varLib.load_masters(designspace, otfFinder)
+    logger.progress("Reading source fonts...")
+    for i, master_font in enumerate(master_fonts):
+        designspace.sources[i].font = master_font
+
+    # Subset source fonts
+    if options.include_glyphs_path:
+        logger.progress("Subsetting source fonts...")
+        subsetDict = getSubset(options.include_glyphs_path)
+        subset_masters(designspace, subsetDict)
+
+    if options.check_compatibility:
+        logger.progress("Checking outline compatibility in source fonts...")
+        font_list = [src.font for src in designspace.sources]
+        default_font = designspace.sources[ds_data.base_idx].font
+        vf = deepcopy(default_font)
+        # We copy vf from default_font, because we use VF to hold
+        # merged arguments from each source font charstring - this alters
+        # the font, which we don't want to do to the default font.
+        do_compatibility(vf, font_list)
+
+    logger.progress("Building VF font...")
+    varFont, varModel, masterPaths = varLib.build(designspace, otfFinder)
```

Comment:
Per Codacy, `masterPaths` is unused, so you could replace this line with something like:
```
 varFont, varModel, _ = varLib.build(designspace, otfFinder)
```
Assuming that the third value returned by `varLib.build()` is actually used elsewhere, otherwise just remove that from what's returned.